realm/realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -105,45 +113,44 @@ public boolean generate() {
 
         if (!categorizeClassElements()) return false;
         if (!checkListTypes()) return  false;
-        if (!checkMethods()) return false;
         if (!checkDefaultConstructor()) return false;
-        if (!checkRequiredGetters()) return false;
-        if (!checkRequireSetters()) return false;
+        if (!checkForFinalFields()) return false;
+        if (!checkForTransientFields()) return false;
+        if (!checkForVolatileFields()) return false;
 
         return true; // Meta data was successfully generated
     }
 
-    // Check that only allowed methods are present in the RealmObject class
-    private boolean checkMethods() {
-        for (ExecutableElement executableElement : methods) {
-            String methodName = executableElement.getSimpleName().toString();
-
-            // Check the modifiers of the method
-            Set<Modifier> modifiers = executableElement.getModifiers();
-            if (modifiers.contains(Modifier.STATIC)) {
-                continue; // We're cool with static methods. Move along!
-            } else if (!modifiers.contains(Modifier.PUBLIC)) {
-                Utils.error("The methods of the RealmObject class must be public", executableElement);
+    private boolean checkForTransientFields() {
+        for (VariableElement field : fields) {
+            if (field.getModifiers().contains(Modifier.TRANSIENT)) {
```

Comment:
shouldn't exclude instead the model from implementing `Serializable`? `transient` without `Serializable` doesn't make sens IMO